Cesium.CodeGen/Contexts/FunctionScope.cs

```diff
@@ -138,4 +138,12 @@ public void MergeScope(BlockScope scope)
             _variableDefinition.Add(variableName, variableDefinition);
         }
     }
+
+    private int tempLocalIndex = 0;
+
+    /// <inheritdoc />
+    public string GetTmpVariable()
+    {
+        return "local_" + tempLocalIndex++;
```

Comment:
I'd suggest prefixing these with `__`, to 100% avoid clash with user-defined variables. (`__`-names are reserved for us according to the language standard.)